compute/cloud-client/src/main/java/compute/disks/CloneEncryptedDiskManagedKey.java

```diff
@@ -0,0 +1,107 @@
+    // "zones/{zone}/diskTypes/(pd-standard|pd-ssd|pd-balanced|pd-extreme)".
+    // For example: "zones/us-west3-b/diskTypes/pd-ssd"
+    String diskType = String.format("zones/%s/diskTypes/pd-ssd", zone);
+
+    // Size of the new disk in gigabytes.
+    int diskSizeGb = 10;
+
+    // A link to the disk you want to use as a source for the new disk.
+    // This value uses the following format:
+    // "projects/{project_name}/zones/{zone}/disks/{disk_name}"
+    String diskLink = String.format("projects/%s/zones/%s/disks/%s", "PROJECT_NAME", "ZONE",
+        "DISK_NAME");
+
+    // URL of the key from KMS. The key might be from another project, as
+    // long as you have access to it. The data will be encrypted with the same key
+    // in the new disk. This value uses following format:
+    // "projects/{kms_project_id}/locations/{region}/keyRings/{key_ring}/cryptoKeys/{key}"
+    String kmsKeyName = "kms-key-name";
+
+    createDiskFromKmsEncryptedDisk(project, zone, diskName, diskType, diskSizeGb, diskLink,
+        kmsKeyName);
+  }
+
+  // Creates a zonal non-boot disk in a project with the copy of data from an existing disk.
+  // The encryption key must be the same for the source disk and the new disk.
+  public static void createDiskFromKmsEncryptedDisk(String project, String zone, String diskName,
+      String diskType, int diskSizeGb, String diskLink, String kmsKeyName)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the `disksClient.close()` method on the client to safely
+    // clean up any remaining background resources.
+    try (DisksClient disksClient = DisksClient.create()) {
+
+      // Create a disk and set the KMS encryption key name.
+      Disk disk = Disk.newBuilder()
+          .setZone(zone)
+          .setName(diskName)
+          .setType(diskType)
+          .setSizeGb(diskSizeGb)
+          .setSourceDisk(diskLink)
+          .setDiskEncryptionKey(CustomerEncryptionKey.newBuilder()
+              .setKmsKeyName(kmsKeyName)
+              .build())
+          .build();
+
+      // Wait for the insert disk operation to complete.
+      Operation operation = disksClient.insertAsync(
+          InsertDiskRequest.newBuilder()
+              .setProject(project)
+              .setZone(zone)
+              .setDiskResource(disk)
+              .build()).get(3, TimeUnit.MINUTES);
+
+      if (operation.hasError()) {
+        System.out.println("Disk creation failed!");
+        throw new Error(operation.getError().toString());
+      }
+      System.out.println(
+          "Disk cloned with KMS encryption key. Operation Status: " + operation.getStatus());
+    }
+  }
+}
+// [END compute_disk_clone_encrypted_disk_kms]
```

Comment:
Add new line at end of file.
```suggestion
// [END compute_disk_clone_encrypted_disk_kms]

```

Reply:
Done.